File: Domain/Message.fs
module Domain.Message

open System

type MessageRequest =
    { Text : string }

type Message =
    { Id : Guid
      Text : string
      CreatedAt : DateTime
      UpdatedAt : DateTime }

    static member CreateMessage(messageRequest : MessageRequest) =
        let newMessage =
            { Id = Guid.Empty
              Text = messageRequest.Text
              CreatedAt = DateTime.MinValue
              UpdatedAt = DateTime.MinValue }
        newMessage
    
    member __.UpdateMessage(messageRequest : MessageRequest) =
      let updatedMessage = { __ with Text = messageRequest.Text }
      updatedMessage

File: Repository/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Repository
{
    public class Message
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        
        public string Text { get; set; }
        
        public DateTime CreatedAt { get; set; }
        
        public DateTime UpdatedAt { get; set; }
    }
}